src/rhsmlib/facts/insights.py

```diff
@@ -61,14 +61,20 @@ def get_insights_machine_id(self):
         else:
             machine_id_filepath = self.DEFAULT_INSIGHTS_MACHINE_ID
 
-        try:
-            with open(machine_id_filepath, "r") as fd:
-                machine_id = fd.read()
-        except IOError as err:
-            log.debug("Unable to read insights machine_id file: %s, error: %s" % (machine_id_filepath, err))
-        else:
-            insights_id = {
-                "insights_id": machine_id
-            }
+        paths_to_check = [
```

Comment:
I would find this easier to read if `paths_to_check` always included `/etc/insights-client/machine-id` and `/etc/redhat-access-insights/machine-id`. Consider setting that first, and then doing `paths_to_check.insert(0, insights_constants.machine_id_file)` when appropriate.